sample-schemas/demo_schema.erl

```diff
@@ -0,0 +1,60 @@
+-module(demo_schema).
+
+-include_lib("typerefl/include/types.hrl").
+
+-type ip() :: ipv4() | ipv6().
+-type ipv4() :: {0..255, 0..255, 0..255, 0..255}.
+-type ipv6() :: {0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535, 0..65535}.
+
+-reflect_type([ip/0]).
+
+-export([keys/0]).
+
+keys() ->
```

Comment:
maybe rename it to:
`fields() ->`
and add another function
```
namespace(source) -> "foo".
namespace(mapping) -> app_foo.
```
so we do not have to repeat `"foo." and `"app_foo."`

Reply:
Sometimes mapping is not one-to-one.

Reply:
that's true.
but the namespace for source is almost always fixed?
if not, at leat we want it to be fixed?

I'd like to see a nested 'object' in the sample schema to trigger more discussion.
on top of my head:

```
fileds("foo") ->
  [{"f1", fun f1/1}];
fields("foo.nested") ->
  [{"nf1", fun nf1/1}].
```

Reply:
not all fields need a mapping target.
if the mapping clause returns `undefined`, it's treated as a value for the wrapping field.
or even a value for later retrieval when need for `translation`.

Reply:
> I'd like to see a nested 'object' in the sample schema to trigger more discussion.

added

Reply:
sorry, bad example. here again:

```
fileds("foo") ->
  [{"f1", fun f1/1},
   {"f2", fun nested/1}
   {"f3", fun array_of_nested/1}
  ];
fields("foo.nested") ->
  [{"nf1", fun nf1/1}].

nested(type) -> fileds("foo.nested");
...

array_field(type) -> array(fields("foo.nested"));
...
```

Reply:
again, I'm not sure how `typerefl` fits in.
should we use `typerefl` only for primitive types ?